content/sensu-core/0.29/guides/troubleshooting.md

```diff
@@ -97,6 +97,68 @@ Sensu staff, or community members may ask to see your logs. You can view them at
 {{< highlight shell >}}
 tail -n 10000 /var/log/sensu/sensu-server.log > sensu-server-10k.log && gzip -9 sensu-server-10k.log{{< /highlight >}}
 
+### Troubleshooting via the Local Client Socket
+This checklist is intended to be a step-by-step walkthrough of how to troubleshoot Sensu via the local client socket. There will be various examples for you to use and build on as you troubleshoot. 
+
+#### Local Client Socket
+By default the sensu-client process listens for check results on a TCP socket. This allows you to submit ad-hoc check results, a capability which is very useful in  troubleshooting issues with Sensu, as well as testing and verifying that a particular configuration works as expected.
+
+This client socket can be disabled, but does provide a few configurable attributes. See the [client reference documentation][13] for further information. 
+
+Before we start, let’s take a look at the prerequisites for using the client socket:
+
+* sensu package installed
+* nc package installed 
+* jq package installed (not required, but will help when reading JSON results)
+* sensu-client has minimal working configuration (client's keepalive timestamp (visible in dashboard or /clients API endpoint) is consistently being updated)
+* sensu-client running (Can be verified with `systemctl status sensu-client`)
+* Local socket is open (Can be verified with `netstat -tnlp | grep 3030` and `nc -vz localhost 3030`)
+
+Once the prerequisites have been met, we can move to troubleshooting.
+
+#### Troubleshooting steps
+
+Consider the following scenario: Sensu has been installed, has been verified to be working correctly (alerts are seen in the dashboard), and is configured to send alerts via the [mailer handler][14]. However, mail doesn't appear to be coming through.
+
+We'll start by crafting a test command to send to the local socket. Why? There are several reasons for using the client socket to troubleshoot:
+
+1. Less configuration overhead. I.e., you don't have to push a check via configuration management.
+2. The ability to issue a check and subsequently resolve it is instantaneous. There is no waiting on a check interval to elapse before the result is published.
+3. The ability to customize what sort
+
+echo ‘{“name”: “testing”, “output”: “THIS IS AN ERROR”, “status”: 2, “refresh”: 10, “handlers”: [“mailer”]}’ | nc localhost 3030
+
+_WARNING: Successfully submitting a check result this way will be indicated by `ok` being printed on the next line -- typically this is appears ahead of the command prompt so it can be easily missed. See below for an example._
+
+{{< highlight shell >}}
+[root@sensu ~]# echo '{"name": "testing_error", "status": 2, "output": "An error event should be created", "refresh": 10, "handlers": [ "mailer"]}' | nc 127.0.0.1 3030
```

Comment:
I'd prefer including `sudo` in the command here or adding a note about requiring privileges. Otherwise, we'll need to change the syntax highlighting to `text` so the `#...` doesn't render as a comment.

Reply:
Shouldn't need elevated privileges for this. Perhaps the prompt should change to `$` or go away completely?